app/controllers/concerns/recorder.rb

```diff
@@ -59,7 +98,8 @@ def filter_recordings(api_res, search)
                ((r[:metadata].nil? || r[:metadata][:name].nil?) &&
                  r[:name].downcase.include?(search)) ||
                r[:participants].include?(search) ||
-               !r[:playbacks].select { |p| p[:type].downcase.include?(search) }.empty?
+               !r[:playbacks].select { |p| p[:type].downcase.include?(search) }.empty? ||
+               (Room.find_by(bbb_id: r[:meetingID]).owner.email.downcase.include?(search) && search_name)
```

Comment:
You might want to flip around the and statement. That way if search_name is false it won't do the search by room owner which should improve the query speed.

Reply:
Done